llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7458,6 +7403,21 @@ LoopVectorizationPlanner::executePlan(
   std::tie(State.CFG.PrevBB, CanonicalIVStartValue) =
       ILV.createVectorizedLoopSkeleton(ExpandedSCEVs ? *ExpandedSCEVs
                                                      : State.ExpandedSCEVs);
+#ifdef EXPENSIVE_CHECKS
+  assert(DT->verify(DominatorTree::VerificationLevel::Fast));
+#endif
+
+  VPBasicBlock *MiddleVPBB =
+      cast<VPBasicBlock>(BestVPlan.getVectorLoopRegion()->getSingleSuccessor());
```

Comment:
Split off a patch that uses VPIRBasicBlock for middle.block in https://github.com/llvm/llvm-project/pull/95816. It might be cleaner to create a regular VPBasicBlock first if the IR basic block hasn't been created yet and later replace it with a VPIRBasicBlock?

Reply:
Replacing VPBB with VPIRBB should also be fine. What happens if the replacement is missing? If IRBB is to be set late, it can be asserted to exist.